### PR TITLE
feat: add lcp field to metadata

### PR DESCRIPTION
**Pull request by azanbinzahid**
## Description:
This PR adds `lead_capture_form_url` to `AdditionalMetadata` model in `course_metadata` app.

## Linked Ticket:
[PROD-2689](https://openedx.atlassian.net/browse/PROD-2689)

## TODO:

- [x] Run migration
- [ ] Add more fields

### CODE REVIEW

File: course_discovery/apps/api/serializers.py [hunk 1]
```diff
@@ -1170,12 +1170,16 @@ def create(self, validated_data):
...
+        lead_capture_form_url = additional_metadata.get('lead_capture_form_url')
 
         if external_url:
             additional_metadata, _ = AdditionalMetadata.objects.get_or_create(external_url=external_url)
             if external_identifier:
                 additional_metadata.external_identifier = external_identifier
                 additional_metadata.save()
+            if lead_capture_form_url:
+                additional_metadata.lead_capture_form_url = lead_capture_form_url
+                additional_metadata.save()
```
Comment (DawoudSheraz) [inline course_discovery/apps/api/serializers.py]: Guess we can take some time to see how this part can be improved, considering we can expect more fields' addition soon.

Comment (DawoudSheraz) [reply course_discovery/apps/api/serializers.py]: We might be able to utilize the instance object to get additional_metadata object, if present.

File: course_discovery/apps/api/serializers.py [hunk 2]
```diff
@@ -1170,12 +1170,16 @@ def create(self, validated_data):
     def update_additional_metadata(self, instance, additional_metadata):
         external_url = additional_metadata and additional_metadata.get('external_url')
         external_identifier = additional_metadata.get('external_identifier')
+        lead_capture_form_url = additional_metadata.get('lead_capture_form_url')
 
         if external_url:
             additional_metadata, _ = AdditionalMetadata.objects.get_or_create(external_url=external_url)
```
Comment (azanbinzahid) [inline course_discovery/apps/api/serializers.py]: - use create or update with instance object 
- use default blank value while getting

File: course_discovery/apps/api/serializers.py [hunk 3]
```diff
@@ -1168,16 +1168,13 @@ def create(self, validated_data):
...
-                additional_metadata.save()
-            instance.additional_metadata = additional_metadata
-            instance.save()
+
+        if instance.additional_metadata:
+            AdditionalMetadata.objects.filter(id=instance.additional_metadata.id).update(**additional_metadata)
+        else:
+            instance.additional_metadata = AdditionalMetadata.objects.create(**additional_metadata)
+
+        # save() will be called by main update()
```
Comment (DawoudSheraz) [inline course_discovery/apps/api/serializers.py:1177]: we can add this comment in the method docstring.

File: course_discovery/apps/api/serializers.py [hunk 4]
```diff
@@ -1168,16 +1168,13 @@ def create(self, validated_data):
...
-            if external_identifier:
-                additional_metadata.external_identifier = external_identifier
-                additional_metadata.save()
-            instance.additional_metadata = additional_metadata
-            instance.save()
+
+        if instance.additional_metadata:
+            AdditionalMetadata.objects.filter(id=instance.additional_metadata.id).update(**additional_metadata)
+        else:
+            instance.additional_metadata = AdditionalMetadata.objects.create(**additional_metadata)
```
Comment (DawoudSheraz) [inline course_discovery/apps/api/serializers.py:1175]: Is this method called when the course is updated from the publisher, considering the publisher is not handling this model? What happens if a course ingested from CSV loader is later updated from Publisher?

**Review by DawoudSheraz — COMMENTED** [review]
- Is the infamous test also failing on the local?

File: course_discovery/apps/api/serializers.py (hunk 4 above)
Comment (azanbinzahid) [reply course_discovery/apps/api/serializers.py:1175]: This method will be triggered by regular update method of API, [ref](https://github.com/openedx/course-discovery/blob/d954033d0b172b35e6c45df4a1f07c4253c39168/course_discovery/apps/api/serializers.py#L1181).

File: course_discovery/apps/api/serializers.py (hunk 3 above)
Comment (azanbinzahid) [reply course_discovery/apps/api/serializers.py:1177]: kinda followed [this](https://github.com/openedx/course-discovery/blob/d954033d0b172b35e6c45df4a1f07c4253c39168/course_discovery/apps/api/serializers.py#L931) convention.

Comment (DawoudSheraz) [reply course_discovery/apps/api/serializers.py:1177]: Ok, but docstring is a better place for such comments.

**Review by DawoudSheraz — APPROVED** [review]
once the build is green.

**Comment by DawoudSheraz** [comment]
Also looping in @ziafazal for a review.

**Review by ziafazal — APPROVED** [review]
👍  Make sure tests are passed.

**Review by uzairr — APPROVED** [review]

**Pull request merged**